lib/mozart_fetcher/timeout_parser.ex

```diff
@@ -0,0 +1,37 @@
+defmodule MozartFetcher.TimeoutParser do
+  @default_timeout MozartFetcher.content_timeout()
+
+  def parse(endpoint) do
+    uri = URI.parse(endpoint)
+    query(uri.query)
+  end
+
+  defp query(nil) do
+    @default_timeout
+  end
+
+  defp query(q) do
+    qs = URI.decode_query(q)
+    fetch(qs["timeout"])
+  end
+
+  defp fetch(nil) do
+    @default_timeout
+  end
+
+  defp fetch("0") do
+    @default_timeout
+  end
+
+  defp fetch(qs_timeout) do
+    parse_timeout(Integer.parse(qs_timeout))
```

Comment:
I like `String.to_integer` more than `Integer.parse` as it simply returns the integer value, not the tuple with the remainder, this can make it easier to handle when piping (which might have looked nicer here than nesting the method calls) as you do need to destructure it but it's only a minor difference. On the downside `Integer.parse` can handle strings with decimals like "2.5" but `String.to_integer` would fail